galleon-plugins/src/main/java/org/wildfly/galleon/plugin/WfInstallPlugin.java

```diff
@@ -233,6 +240,20 @@ public void postInstall(ProvisioningRuntime runtime) throws ProvisioningExceptio
         log = runtime.getMessageWriter();
         log.verbose("WildFly Galleon Installation Plugin");
 
+        if (runtime.isOptionSet(OPTION_RECORD_ARTIFACTS)) {
+            final String pathValue = runtime.getOptionValue(OPTION_RECORD_ARTIFACTS);
+            if (!StringUtils.isEmpty(pathValue)) {
+                try {
+                    log.verbose("Starting artifact log");
+                    artifactRecorder = Optional.of(new ArtifactRecorder(runtime.getStagedDir(), Path.of(pathValue)));
+                } catch (IOException e) {
+                    throw new ProvisioningException("Unable to create artifact.log", e);
+                }
+            }
```

Comment:
@spyrkob , we would get NPE in case the pathValue is empty. Should we have a default value or fail if the path is empty?

Reply:
@jfdenise if the pathValue is empty the artifactRecorder is not going to be initiated so it'll be treated as if the option was no set.

Default value would be good though - is `.installation/cache` OK?